docs/java-tutorial-1-giraph.md

```diff
@@ -0,0 +1,96 @@
+
+## Try some example giraph apps
+
+We provide some example giraph algorithms, i.e. SSSP, PageRank in [grape-demo.jar](https://graphscope.oss-cn-beijing.aliyuncs.com/jar/grape-demo-0.18.0-shaded.jar).
+You can try to run these Giraph algorithms on GraphScope.
+
+As `Giraph` allows user to load graph with customized loader, we support `Giraph VertexInputFormat` and `Giraph EdgeInputFormat` with `session.load_from` method.
+```python
+vformat = "giraph:com.alibaba.graphscope.example.giraph.format.P2PVertexInputFormat"
+eformat = "giraph:com.alibaba.graphscope.example.giraph.format.P2PEdgeInputFormat"
+
+#clone https://github.com/GraphScope/gstest to GS_TEST_DIR
+graph = graphscope_session.load_from(
+    vertices="/path/to/vertex-input",
+    vformat=vformat,
+    edges="/path/to/edge-input",
+    eformat=eformat,
+)
+```
+
+vertices and edges should points to vertex input and edge input. We also provide some example dataset `gstest` at [Graphscope/gstest](https://github.com/7br/gstest.git). You can use this sample dataset by:
+
+```bash
+git clone -b master --single-branch --depth=1 https://github.com/7br/gstest.git ${GS_TEST_DIR}
+```
+
+{GS_TEST_DIR} denote the path where `gstest` dataset resides. Then you can load graph via graphscope python client, and query the graph with giraph app.
+
+```python
+import graphscope
+
+"""Or lauch session in k8s cluster"""
+sess = graphscope.session(cluster_type='hosts') 
+
+graphscope_session.add_lib("path/to/grape-demo.jar")
+
+vformat = "giraph:com.alibaba.graphscope.example.giraph.format.P2PVertexInputFormat"
+eformat = "giraph:com.alibaba.graphscope.example.giraph.format.P2PEdgeInputFormat"
+
+#clone https://github.com/GraphScope/gstest to GS_TEST_DIR
+graph = graphscope_session.load_from(
+    vertices=os.path.expandvars("${GS_TEST_DIR}/p2p-31.v"),
+    vformat=vformat,
+    edges=os.path.expandvars("${GS_TEST_DIR}/p2p-31.e"),
+    eformat=eformat,
+)
+graph = graph._project_to_simple(v_prop="vdata", e_prop="data")
+
+giraph_sssp = load_app(algo="giraph:com.alibaba.graphscope.example.giraph.SSSP")
+ctx = giraph_sssp(g, sourceId=6)
+```
+
+## Run your own Giraph apps.
+
+After a successful running of example giraph SSSP algorithm, you may want to try your own giraph algorithm on GraphScope(which runs much faster then Giraph itself). 
+
+### Develop Giraph algorithm
+
+You can implement your algorithm towards Giraph' original API. Although almost all APIs are supported, there are indeed some limitation of Giraph-on-Graphscope.
+
+- Currently graph modification API is not supported.
+- Using of Complex Writable will cause performance degradation.
+
+### Submit to GraphScope.
+
+The procedure almost the same as above, except that you need to replace the submitted jar, and choose right `InputFormat` classes.
+
+```python
+import graphscope
+
+"""Or lauch session in k8s cluster"""
+sess = graphscope.session(cluster_type='hosts') 
+
+graphscope_session.add_lib("path/to/grape-demo.jar")
```

Comment:
I just wonder if this is a path on local? or on the cluster?

Reply:
Path on local. Coordinator will distribute the jar to cluster.

Reply:
Sure, maybe speak it out